styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -400,7 +404,7 @@ public void create(Environment environment) {
     startRuntimeConfigUpdate(styxConfig, tickExecutor, dequeueRateLimiter);
     startCleaner(cleaner, tickExecutor);
 
-    setupMetrics(stateManager, workflowCache, storage, dequeueRateLimiter, stats);
+    setupMetrics(queuedStateManager, workflowCache, storage, dequeueRateLimiter, stats);
```

Comment:
With this we'll miss tracing of calls made to QueuedStateManager for getting metrics. Is that intended? Can we send the proxy instead?

Reply:
The `setupMetrics` method needs to access the `QueuedStateManager` directly, it cannot use the `StateManager` interface.

Reply:
👍 sounds like it should be refactored. but that can go in another PR then